rumqttc/src/lib.rs

```diff
@@ -412,6 +414,13 @@ impl NetworkOptions {
     pub fn connection_timeout(&self) -> u64 {
         self.conn_timeout
     }
+
+    /// bind connection to a specific network device by name
+    /// only supported on linux, android and fuchsia
+    pub fn set_bind_device(&mut self, bind_device: &str) -> &mut Self {
+        self.bind_device = Some(bind_device.to_string());
+        self
+    }
```

Comment:
would it make sense to set this value if `target_os` is not one of the supported OS?
 
i would prefer to add `cfg` in this method as well as in `bind_device` member in the struct

Reply:
Yeah, I considered that also, but decided that keeping it in it would make any examples that use the function more portable albeit non-functional.
Feels messy putting `#[cfg(all(any(target_os = "android", target_os = "fuchsia", target_os = "linux")))]` all over the place. What do you think about if I add a `bind-device` feature, that `compile_error!`s if it's not the supported os like: https://doc.rust-lang.org/cargo/reference/features.html#mutually-exclusive-features ?

Reply:
Yeah, and also at `NetworkOptions::new()`. I'll add that in.